design/one-pager-default-resource-class.md

```diff
@@ -88,66 +116,237 @@ spec:
+The `Policy` method continues to enable default class references, but
+strongly-typed resource classes introduce a reduction in portability of resource
+claims. Previously, claims could reference a generic `ResourceClass` by `name`
+and `namespace`, and could be moved across providers as long as there existed a
+class with that `name`. Now, claims must omit a `classRef` and rely on the
+existence of a `Policy` to achieve portability.
 
-In this scenario, a default resource class may look like this:
+In addition, becauase claims must reference resource classes using their full
+group, version, and kind, meaning that the creator of a claim forfeits the
+ability to select a *generic* "class of service" (i.e. `mysql-large`).
 
 ```yaml
-apiVersion: core.crossplane.io/v1alpha1
-kind: ResourceClass
+apiVersion: database.crossplane.io/v1alpha1
+kind: MySQLInstance
 metadata:
-  name: cloud-nosql
-  namespace: crossplane-system
-parameters:
-  ...
-defaultForClaimKinds:
-- nosqlinstance.database.crossplane.io
-- dynamodbinstance.storage.crossplane.io
-provisioner: dynamodbinstance.database.aws.crossplane.io/v1alpha1
-providerRef:
-  name: aws-provider
-reclaimPolicy: Delete
+  name: mysql-claim
+spec:
+  classRef:
+    kind: RDSInstanceClass
+    apiVersion: database.aws.crossplane.io/v1alpha1
+    name: standard-mysql
+    namespace: crossplane-system
+  writeConnectionSecretToRef:
+    name: rdsmysql
+  engineVersion: "5.6"
 ```
 
-And a resource claim that defaulted to the resource class above could look like this in the general case:
+To continue to provide the same level of portability for claims that was
+originally present, two enhancements to the `Policy` model can be made:
+
+1. Allow for multiple policies per namespace.
+1. Denote the default policy via label.
+
+### Multiple Policies Per Namespace
+
+Currently, default class controllers, which act on claims that omit a
+`classRef`, will fail to reconcile if multiple `Policies` exist within the
+claim's namespace. For example, if two `MySQLInstancePolicies` are created in
+namespace `crossplane-system`, then a `MySQLInstance` claim created in
+`crossplane-system` that omits a `classRef` will not be assigned either one of
+the classes referenced by the two `MySQLInstancePolicy` objects respectively.
+
+In order to reintroduce the ability to select a class of service in claim, the
+following steps can be taken:
+
+1. Update all `Policy` types to include a `classRef` field instead of a
+   `defaultClassRef` field. This would require updating the `crossplane-runtime`
+   embeddable struct
+   [`Policy`](https://github.com/crossplaneio/crossplane-runtime/blob/e4d61ee2805af680baf16fc2a1d8f79538d0f9bb/apis/core/v1alpha1/resource.go#L93),
+   then bumping the dependency in core Crossplane.
+1. Remove the `classRef` field from all claims entirely. This would require the
+   same steps as above but with
```

Comment:
It seems as though the motivation for removing `classRef` is to force claims to **always** be portable (in the sense that there will never be references to a strongly-typed Thing which is provider-specific). If this is the case, I would suggest that we also state that as one of the goals, instead of simply saying we want the level of portability that we had previously.

Here is the same thing as a question: for the stated goals, do we really need to remove the `classRef` field? What is the motivation for doing it this way instead of just adding the support for the policy ref?

Reply:
I had this same question, but @prasek had some good arguments as to why it should be removed. I will let him provide further context.